CMakeLists.txt

```diff
@@ -49,9 +49,15 @@ endif()
 set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${CMAKE_C_FLAGS_RELEASE}")
 
 if (TARGET_PLATFORM STREQUAL "rdkb")
-        set(CMAKE_SKIP_RPATH TRUE)
+    set(CMAKE_SKIP_RPATH TRUE)
 endif()
 
+list (FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}" isSystemDir)
+if ("${isSystemDir}" STREQUAL "-1")
+    set (CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
+endif ("${isSystemDir}" STREQUAL "-1")
+set (CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
```

Comment:
According to the "Common questions" section in the referenced wiki page, this one line should be enough. And I think that also [`CMAKE_INSTALL_RPATH_USE_LINK_PATH`](https://cmake.org/cmake/help/v3.0/variable/CMAKE_BUILD_WITH_INSTALL_RPATH.html?highlight=cmake_build_with_install_rpath) needs to be set to TRUE. But I don't think all the rest is needed.

Reply:
I tried setting only `CMAKE_INSTALL_RPATH_USE_LINK_PATH` and got 
```
]$ ./build/install/bin/tests/common/poller_test 
./build/install/bin/tests/common/poller_test: error while loading shared libraries: mapfcommon.so.1: cannot open shared object file: No such file or directory
```

So I guess it is needed...

Also, see nng which has a pretty decent cmake support, this is done the same there: https://github.com/nanomsg/nng/blob/master/CMakeLists.txt